File: algorithm/java/크레인인형뽑기게임.java
import java.util.*;

class Solution {
    public int solution(int[][] board, int[] moves) {
        int answer = 0;
        ArrayList<Integer> bucket = new ArrayList<>();
        
        for (int i : moves) {
            for (int j=0; j < board.length; j++) {
                if (board[j][i-1] != 0) {
                    bucket.add(board[j][i-1]);
                    board[j][i-1] = 0;
                    break;
                }
            }
            if (bucket.size() >= 2 && (bucket.get(bucket.size()-1) == bucket.get(bucket.size()-2))) {
                    bucket.remove(bucket.size()-1);
                    bucket.remove(bucket.size()-1);
                    answer += 2;
            }
        }
        
        return answer;
    }
}

File: algorithm/python/크레인인형뽑기게임.py
def solution(board, moves):
    doll = 0
    bucket = []

    for j in moves:
        for i in range(len(board)):
            if board[i][j-1] != 0:
                bucket.append(board[i][j-1])
                board[i][j-1] = 0
                break
    
        if len(bucket) >= 2 and bucket[-1] == bucket[-2]:
            bucket.pop(-1)
            bucket.pop(-1)
            doll += 2

    return doll